src/patch.rs

```diff
@@ -29,11 +29,16 @@
 
 use std::io;
 use std::io::Read;
+use std::ops::DerefMut;
 
 /// Apply a patch to an "old" file, returning the "new" file.
 ///
 /// `old` is the old file, `patch` will be read from with the patch,`new` is the buffer that will be written into.
```

Comment:
Thanks for the PR!

Please add documentation and perhaps an example that it works with a `&mut Vec`, because the trait bounds don't make that obvious.